webpack.production.config.js

```diff
@@ -22,20 +27,15 @@ module.exports = {
             }
         }),
         new webpack.optimize.OccurrenceOrderPlugin(),
-        new webpack.optimize.DedupePlugin()
     ],
 
-    resolve: {
-        extensions: ["", ".webpack.js", ".web.js", ".js"]
-    },
-
     module: {
+        rules: [
+            // All files with a '.ts' or '.tsx' extension will be handled by 'awesome-typescript-loader'.
+            { test: /\.tsx?$/, loader: "awesome-typescript-loader" },
```

Comment:
Honestly I'm not sure I understand webpack enough to answer your question. This is copied from some references in how to get webpack to bundle TypeScript directly from sources (instead of from the built files).